src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -34,6 +34,21 @@ def process_deployment_create_namespace(namespace):
     _validate_deployment_name(namespace)
 
 
+def process_deployment_group_create_namespace(namespace):
+    if bool(namespace.on_error_type):
+        if namespace.on_error_type.lower() == 'specificdeployment':
+            if not bool(namespace.on_error_name):
+                raise CLIError('OnErrorDeployment using SpecificDeployment type should have a deployment name.')
+        elif namespace.on_error_type.lower() == 'lastsuccessful':
+            if bool(namespace.on_error_name):
+                raise CLIError('OnErrorDeployment using LastSuccessful type shouldn not have a deployment name.')
+        else:
+            raise CLIError('Unable to parse type: {}, valid values are SpecificDeployment and LastSuccessful'.format(
+                namespace.on_error_type))
+
+    process_deployment_create_namespace(namespace)
```

Comment:
You can revert this since you will set the type value based on whether a name is specified or not.